br/pkg/lightning/backend/local/local.go

```diff
@@ -1342,9 +1342,16 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID, regi
 		log.L().Info("engine contains no kv, skip import", zap.Stringer("engine", engineUUID))
 		return nil
 	}
-	regionSplitKeys := int64(regionMaxKeyCount)
-	if regionSplitSize > defaultRegionSplitSize {
-		regionSplitKeys = int64(float64(regionSplitSize) / float64(defaultRegionSplitSize) * float64(regionMaxKeyCount))
+	kvRegionSplitSize, regionSplitKeys, err := getRegionSplitSizeKeys(ctx, local.pdCtl.GetPDClient(), local.tls)
+	if err == nil {
+		if kvRegionSplitSize > regionSplitSize {
+			regionSplitSize = kvRegionSplitSize
+		}
+	} else {
+		regionSplitKeys = int64(regionMaxKeyCount)
+		if regionSplitSize > defaultRegionSplitSize {
+			regionSplitKeys = int64(float64(regionSplitSize) / float64(defaultRegionSplitSize) * float64(regionMaxKeyCount))
+		}
```

Comment:
I think it's too risky to use the huge split keys and split size by default.

Reply:
addressed in https://github.com/pingcap/tidb/pull/34537/commits/0887fdbca612413ef784667b7b9fc74f44fd872d
PTAL